packages/common/src/api/tan-query/useCollectionByParams.ts

```diff
@@ -0,0 +1,69 @@
+import { useEffect } from 'react'
+
+import { useDispatch } from 'react-redux'
+
+import { ID } from '~/models/Identifiers'
+import {
+  collectionPageActions,
+  collectionPageLineupActions
+} from '~/store/pages'
+
+import { Config } from './types'
+import { useCollection } from './useCollection'
+import { useCollectionByPermalink } from './useCollectionByPermalink'
+
+type CollectionParams =
+  | { collectionId?: ID }
+  | { handle?: string; slug?: string; collectionType?: 'album' | 'playlist' }
```

Comment:
we have a CollectionType in saved-collections. It's pluralized though. Maybe can move it out or make another one for this?